chapters/en/slides/module6/module6_02.Rmd

```diff
@@ -0,0 +1,352 @@
+or and exact hex-code.
+
+Here we also make the stroke width slightly larger 
+so that it is easier to see
+the country borders.
+
+---
+
+## We can zoom and pan the map
+
+```{python}
+alt.Chart(world_map).mark_geoshape(
+    color='#2a1d0c', stroke='#706545', strokeWidth=0.5
+).project(type='equalEarth', scale=500, translate=[140, 610])
+```
+
+Notes:
+By default Altair automatically adjusts the projection
+so that all the data fits within the width and height of the chart.
+
+We can change this via the projections parameters `scale` (zoom level) and `translate` (panning).
+Here we adjust them to zoom in and center the map on Europe.
+Getting the panning right when zooming in can take a bit of fiddling.
+
+---
+
+## Working with individual countries can show regional details
+
+```{python}
+state_map = alt.topo_feature(data.us_10m.url, 'states')
+alt.Chart(state_map).mark_geoshape().project(type='equalEarth')
+```
+
+Notes:
+If we want information on regions within a country,
+we have to use a geo/topojson file that contains this info.
+There is already one in the vega demo datasets for the US,
+which is the one we are using here.
+
+The loading of the data is exactly the same as previously,
+but as you can see,
+the `equalEarth` projection is not very suitable for a single country.
+
+The US it is very small and both Alaska and Hawaii 
+are shown far away from the rest of the country.
+
+---
+
+## There are special projections for some countries such as the US
+
+```{python}
+alt.Chart(state_map).mark_geoshape().project(type='albersUsa')
+```
+
+Notes:
+There special US projection `albersUsa`
+is much preferred when only showing this country.
+You might recognize the projection from maps that you have seen elsewhere
+such as on the news.
+Characteristic is the placement of Alaska and Hawaii
+close to the rest of the country.
+
+---
+
+## Coloring individual regions differently
+
+```{python}
+(alt.Chart(state_map).mark_geoshape().encode(
+    color='id:Q')
+.project(type='albersUsa'))
+```
+
+Notes:
+So far we have seen how to visualize region in a uniform color,
+but often we want to use color to represent amounts within regions
+so that we can contrast them with each other.
+How can we do this in Altair?
+
+As you can see in this slide,
+we can use `encode` together with `mark_geojson`,
+the same way we have used it with the other marks.
+This type of visualization can be thought of as a geographical heatmap
+and is often referred to as a "choropleth map".
+
+The `'id'` field here comes from the sample topojson file.
+If we opened up and looked at this file,
+we would find that it does contains a field called `'id'`,
+which is a numerical identifier for each state 
+in alphabetical order.
+
+Because we are not using a pandas dataframe,
+we mush specify the type of data in the column.
+Here we are specifying it a quantitative for demonstration purposes
+because the legend would become very long
+if we specified it as ordinal or nominal.
```

Comment:
Do we even want this slide? We would never do this in real life (map id to the geo region). Is it because you want to show how to get data from the topoJSON file? If not, then I suggest we remove this slide and just go straight to the putting data from a pandas data frame on this map.

One case where there is open data where we would want to do this is earthquake data from the USGS. The geoJSON files there contain richer info that you cannot get in a csv. See: https://earthquake.usgs.gov/earthquakes/search/

```suggestion
So far we have seen how to visualize regions with a uniform colour,
but often we want to use colour to map values or summary statistics
from our data to different geographic regions. How can we do this in Altair?

As you can see in this slide,
we can use `encode` together with `mark_geojson`,
the same way we have used it with the other marks.
This type of visualization can be thought of as a geographical heatmap
and is often referred to as a "choropleth map".

To introduce how we can do this - we will map the states identification
number from our topoJSON file we loaded, to the corresponding 
geographic region for each state. Normally, we would not do this, 
but instead map some other data to each geographic region. We will
do that in the following slides.

The `'id'` field here comes from the sample topoJSON file.
If we opened up and looked at this file,
we would find that it contains a field called `'id'`,
which is a numerical identifier for each state 
in alphabetical order.

Because we are not using a pandas dataframe,
we mush specify the type of data in the column.
Here we are specifying it as a quantitative for demonstration purposes
because the legend would become very long
if we specified it as ordinal or nominal.
```

Reply:
Yeah I think it helps understand both that geojson can contain data that we want to visualize and where the ID is coming from when we are using it for linking to the other datasetin the next slide